tests/test_scrapy_settings.py

```diff
@@ -0,0 +1,78 @@
+import attr
+from pytest_twisted import inlineCallbacks
+from scrapy import Request, Spider
+from scrapy.settings import Settings
+
+from scrapy_poet.page_input_providers import (
+    PageObjectInputProvider,
+    provides,
+)
+
+from tests.utils import crawl_single_item, HtmlResource
+
+
+class ProductHtml(HtmlResource):
+
+    html = """
+    <html>
+        <div class="breadcrumbs">
+            <a href="/food">Food</a> / 
+            <a href="/food/sweets">Sweets</a>
+        </div>
+        <h1 class="name">Chocolate</h1>
+        <p>Price: <span class="price">22€</span></p>
+        <p class="description">The best chocolate ever</p>
+    </html>
+    """
+
+
+@attr.s(auto_attribs=True)
+class MySettings:
+
+    robotstxt_obey: bool
+
+
+@provides(MySettings)
+class MySettingsProvider(PageObjectInputProvider):
```

Comment:
It would be nice to have also a test with a provider which uses an injectable and inside the settings. For example:

```py

@attr.s(auto_attribs=True)
class SettingsWrapper(Injectable):
    settings: Settings

@provides(MySettings)
class MySettingsProvider(PageObjectInputProvider):
 
    def __init__(self, settings_wrapper: SettingsWrapper):
        self.settings = settings_wrapper.settings
```

Just to check that full recursive injection is also working.

Reply:
I've just refactored tests for better readability and to make sure recursive injection is working.